python.d/unbound.chart.py

```diff
@@ -0,0 +1,158 @@
+    'total.recursion.time.avg': 'recursive_avg',
+    'total.recursion.time.median': 'recursive_med',
+    'msg.cache.count': 'cache_message',
+    'rrset.cache.count': 'cache_rrset',
+    'infra.cache.count': 'cache_infra',
+    'key.cache.count': 'cache_key',
+    'dnscrypt_shared_secret.cache.count': 'cache_dnscss',
+    'dnscrypt_nonce.cache.count': 'cache_dnscn'
+}
+
+
+class Service(SocketService):
+    def __init__(self, configuration=None, name=None):
+        # The unbound control protocol is always SSL encapsulated
+        # unless it's used over a UNIX socket, so enable SSL _before_
+        # doing the normal SocketService initialization.
+        configuration['ssl'] = True
+        self.port = 8935
+        SocketService.__init__(self, configuration, name)
+        self.ext = self.configuration.get('extended', None)
+        self.ubconf = self.configuration.get('ubconf', '/etc/unbound/unbound.conf')
+        self.order = ORDER
+        self.definitions = CHARTS
+        if self.ext:
+            self.order = self.order + EXTENDED_ORDER
+            self.definitions.update(EXTENDED_CHARTS)
+        self.request = 'UBCT1 stats\n'
+        self._parse_config()
+        self.debug('Unbound config: {0}'.format(self.ubconf))
+        if os.access(self.ubconf, os.R_OK):
+            with open(self.ubconf, 'r') as ubconf:
+                try:
+                    conf = yaml.load(ubconf)
+                except yaml.YAMLError:
+                    conf = dict()
+            if self.ext is None:
+                if 'extended-statistics' in conf['server'].keys():
+                    self.ext = conf['server']['extended-statistics']
+            if 'remote-control' in conf.keys():
+                if conf['remote-control'].get('control-use-cert', False):
+                    if not self.key:
+                        self.key = conf['remote-control'].get('control-key-file', '/etc/unbound/unbound_control.key')
+                    if not self.cert:
+                        self.cert = conf['remote-control'].get('control-cert-file', '/etc/unbound/unbound_control.pem')
+                    self.port = conf['remote-control'].get('control-port', 8953)
+                else:
+                    if not self.unix_socket:
+                        self.unix_socket = conf['remote-control'].get('control-interface')
+        self.debug('Extended stats: {0}'.format(self.ext))
+        if self.unix_socket:
+            self.debug('Using unix socket: {0}'.format(self.unix_socket))
+            for key in self.definitions.keys():
+                self.definitions[key]['options'][4] = 'Local'
+        else:
+            self.debug('Connecting to: {0}:{1}'.format(self.host, self.port))
+            self.debug('Using key: {0}'.format(self.key))
+            self.debug('Using certificate: {0}'.format(self.cert))
+            for key in self.definitions.keys():
+                self.definitions[key]['options'][4] = self.host
```

Comment:
Huge `__init__` is a bad thing. Please move all logic to separate method(s)/check

Reply:
normally I would agree, and I actually originally had everything from line 105 to 123 in a separate method, but I was having significant trouble getting it to work and it was just easier to push it into `__init__`.  I intend to get that taken care of, I just wanted to get the current (mostly working other than the thing below about `float` conversion) code out there for some extra review (and I'm glad I did).